pywinusb/hid/core.py

```diff
@@ -613,16 +612,15 @@ def close(self):
             hid_dll.HidD_FreePreparsedData(ptr_preparsed_data)
 
         # wait for the reading thread to complete before closing device handle
-        while self.__reading_thread and self.__reading_thread.is_active():
-            time.sleep(0.050) # 50 ms latency, just to avoid CPU consumption
+        if self.__reading_thread:
+            self.__reading_thread.join()
 
         if self.hid_handle:
             winapi.CloseHandle(self.hid_handle)
         
         # make sure report procesing thread is closed
         if self.__input_processing_thread:
-            while self.__input_processing_thread.is_alive():
-                time.sleep(0.050)
+            self.__input_processing_thread.join()

```

Comment:
If the assert mentioned below is removed then this will trigger a RuntimeError if USB is disconnected.  This is because InputReportReaderThread calls close and tries to join itself.  PR #25 fixes this problem by removing the call to close from InputReportReaderThread.